db/view/view.cc

```diff
@@ -471,22 +509,65 @@ deletable_row& view_updates::get_view_row(const partition_key& base_key, const c
             } else {
                 computed_value = cdef.get_computation().compute_value(*_base, base_key);
             }
-            return managed_bytes_view(linearized_values.emplace_back(*computed_value));
+
+            return {managed_bytes_view(linearized_values.emplace_back(std::move(computed_value)))};
         }
         switch (base_col->kind) {
         case column_kind::partition_key:
-            return base_key.get_component(*_base, base_col->position());
+            return {base_key.get_component(*_base, base_col->position())};
         case column_kind::clustering_key:
-            return update.key().get_component(*_base, base_col->position());
+            return {update.key().get_component(*_base, base_col->position())};
         default:
             auto& c = update.cells().cell_at(base_col->id);
             auto value_view = base_col->is_atomic() ? c.as_atomic_cell(cdef).value() : c.as_collection_mutation().data;
-            return value_view;
+            return {managed_bytes_view{value_view}};
         }
     });
-    auto& partition = partition_for(partition_key::from_range(_view->partition_key_columns() | get_value));
-    auto ckey = clustering_key::from_range(_view->clustering_key_columns() | get_value);
-    return partition.clustered_row(*_view, std::move(ckey));
+
+    std::vector<view_updates::view_row_entry> ret;
+
+    std::vector<std::vector<bytes_view_with_action>> pk_elems, ck_elems;
+    boost::copy(_view->partition_key_columns() | get_value, std::back_inserter(pk_elems));
+    // If no collection column was found, each of the actions will contain no_action,
+    // in particular, it does not harm to use column 0.
+    size_t action_column = collection_column_position.value_or(0);
+    // Allow for at most one collection computed column in pk and in ck.
+    collection_column_position.reset();
+    boost::copy(_view->clustering_key_columns() | get_value, std::back_inserter(ck_elems));
+
+
+    auto cartesian_product_pk = cartesian_product(pk_elems),
+         cartesian_product_ck = cartesian_product(ck_elems);
```

Comment:
I was sure the comma operator was forbidden in Geneva conventions

Reply:
This is not a comma operator, but a declaration. Therefore your comment is irrelevant.